### ISSUE TITLE
Batch Process for Detoxify Classification

### ISSUE THREAD

**Issue by Nil-Andreu** [LAION-AI/Open-Assistant#504]
I am wondering if for the messages that we were not able to classify, If we could create a batch process to classify them.
Or if maybe we are not interested in storing this information after the conversation.

**Comment by andreaskoepf** [comment]
> I am wondering if for the messages that we were not able to classify, If we could create a batch process to classify them.

Yes, we need a function that retries the detoxify web-call for all messages with a null detoxify-result. A background-thread sould be launched on start-up to do this and then run periodically every few hours.

> Or if maybe we are not interested in storing this information after the conversation.

We want to store the detoxify-results for all messages in the DB, i.e. when a call initially failed we will retry it later through the "batch process".

**Comment by Nil-Andreu** [comment]
Okay thanks! I have to do a couple tasks previous to be able to take this one (f.e. including this classification in the flow of the conversation).

And then I think I could onboard this.

For running periodically‚ why might want to use some sort of task queue with f.e. redis and a worker?
